modelling/src/test/java/org/axonframework/modelling/annotation/InjectEntityParameterResolverFactoryConfigurationEnhancerTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.annotation;
+
+import org.axonframework.configuration.DefaultComponentRegistry;
+import org.axonframework.configuration.LifecycleRegistry;
+import org.axonframework.configuration.NewConfiguration;
+import org.axonframework.eventhandling.ConcludesBatchParameterResolverFactory;
+import org.axonframework.messaging.annotation.MultiParameterResolverFactory;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.junit.jupiter.api.*;
+import org.mockito.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class InjectEntityParameterResolverFactoryConfigurationEnhancerTest {
+
+    @Test
+    void addsInjectEntityParameterResolverFactory() {
+
+        DefaultComponentRegistry componentRegistry = new DefaultComponentRegistry();
+        componentRegistry.disableEnhancerScanning()
+                         .registerEnhancer(new InjectEntityParameterResolverFactoryConfigurationEnhancer())
+                         .registerComponent(ParameterResolverFactory.class,
+                                            (c) -> new ConcludesBatchParameterResolverFactory());
```

Comment:
Now that I am looking at this, I am thinking of the approach you took for the `StateManager`: if there is no parent, simply register the component from the decorator. Why aren't we doing that for the `ParameterResolverFactory`?

Reply:
I don't understand the question, as the approoach of the `HierarchicalParameterResolverFactoryConfigurationEnhancer` is exactly te same as for the Statemanager:

```java
Optional<ParameterResolverFactory> parentComponent = Optional
        .ofNullable(config.getParent())
        .flatMap(p -> p.getOptionalComponent(ParameterResolverFactory.class));
if (parentComponent.isPresent()) {
    return HierarchicalParameterResolverFactory.create(
            parentComponent.get(),
            component
    );
}
return component;

Could you elaborate?

Reply:
This test disables all enhancers, which is why I needed to register a base ParameterResolverFactory here.
I used the conclude batch one, as it's reliably always one. The ClassPath one can change with iterations in the future, as we may add more.

Reply:
Well, I mean that the act of invoking `.registerEnhancer(new InjectEntityParameterResolverFactoryConfigurationEnhancer())` will, indeed, cause issues.

Why doesn't the registered enhancer, using the utility you made, register some form of base parameter resolver itself instead of failing in absence of one?

Reply:
Yes, you are right. I have adjusted the `ParameterResolverFactoryUtils` to either use a decorator, or if not registered yet, register a component. Easy, and removes that burden from an enhancer that can be cancelled.

This is what you meant right? Thanks for the tip! If you're okay with the last commit we can merge this.